File: src/main/java/com/upokecenter/mail/IdentityEncoder.java
package com.upokecenter.mail;
/*
Written by Peter O.
Any copyright to this work is released to the Public Domain.
In case this is not possible, this work is also
licensed under the Unlicense: https://unlicense.org/

 */

import java.io.*;

import com.upokecenter.util.*;
import com.upokecenter.text.*;

  /**
   * This is an internal API.
   */
  final class IdentityEncoder implements ICharacterEncoder {
    public int Encode(int c, IWriter s) {
      if (s == null) {
        throw new NullPointerException("s");
      }
      if (c < 0) {
        return -1;
      }
      c &= 0xff;
      s.write((byte)c);
      return 1;
    }
  }

File: PeterO/Mail/IdentityEncoder.cs
/*
Written by Peter O. in 2014.
Any copyright is dedicated to the Public Domain.
http://creativecommons.org/publicdomain/zero/1.0/
If you like this, you should donate to Peter O.
at: http://upokecenter.dreamhosters.com/articles/donate-now-2/
 */
using System;
using System.Text;
using System.IO;

using PeterO;
using PeterO.Text;

namespace PeterO.Mail {
    /// <include file='../../docs.xml'
    /// path='docs/doc[@name="T:PeterO.Mail.IdentityEncoder"]/*'/>
  internal sealed class IdentityEncoder : ICharacterEncoder
  {
    public int Encode(int c, IWriter s) {
      if (s == null) {
  throw new ArgumentNullException("s");
}
      if (c < 0) {
 return -1;
}
      c &= 0xff;
      s.WriteByte((byte)c);
      return 1;
    }
  }
}